### PR TITLE
MGMT-23770: fix publish_templates for empty API responses

### PR DESCRIPTION
**Pull request by zszabo-rh**
## Summary

- Fix `publish_templates` role failing when the fulfillment-service API returns empty responses (no `size` field)
- Replace `.json.size > 0` with `.json.get('items', []) | length > 0` in all 3 task files
- Add unit tests with a mock HTTP server covering empty, populated, and edge-case responses

## Root Cause

The fulfillment-service uses proto3 `optional int32` for the `size` field. When there are no items, proto3 JSON serialization omits zero-valued optional fields entirely, so the response is `{}` instead of `{"size": 0, "items": []}`. The Ansible conditional `.json.size` then fails with `'dict object' has no attribute 'size'`.

The fix uses `.json.get('items', [])` (Python dict method) rather than dot or bracket notation, because `items` is also a built-in dict method in Python — both `.json.items` and `.json['items']` resolve to `dict.items()` when the key is absent.

## Files Changed

**Bug fix** (3 files, 1 line each):
- `collections/.../publish_templates/tasks/clusters.yaml`
- `collections/.../publish_templates/tasks/compute_instances.yaml`
- `collections/.../publish_templates/tasks/network_classes.yaml`

**Unit tests** (2 new files):
- `collections/.../publish_templates/tests/test.yml` — 3 test scenarios
- `collections/.../publish_templates/tests/mock_api_server.py` — mock HTTP server

## Test plan

- [x] `ansible-lint` passes (0 failures, 0 warnings)
- [x] `pre-commit run --all-files` passes
- [x] Unit test: empty API response (the [MGMT-23770](https://redhat.atlassian.net/browse/MGMT-23770) bug scenario)
- [x] Unit test: populated response (existing items PATCHed, new POSTed)
- [x] Unit test: no `items` key edge case
- [x] E2E on hypershift1 (osac-zszabo): publish_templates playbook against real fulfillment-service with empty network_classes and compute_instance_templates endpoints

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved template publication checks to reliably detect existing items across varied API JSON shapes, ensuring create vs update actions run correctly for clusters, compute instances, and network classes.

* **Tests**
  * Added an end-to-end test playbook and a local mock API server to validate populated, empty, and missing-items responses and expected POST vs PATCH behavior.

### CODE REVIEW

File: collections/ansible_collections/osac/service/roles/publish_templates/tests/test.yml [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        url: "http://127.0.0.1:{{ test_mock_port }}/_calls"
+        return_content: true
+      register: call_log
+      when: test_populated is defined
+
+    - name: Verify PATCH calls for existing items and POST for new
+      ansible.builtin.assert:
+        that:
+          - call_log.json | selectattr('method', 'equalto', 'PATCH') | list | length == 3
+          - call_log.json | selectattr('method', 'equalto', 'POST') | list | length == 2
```
Comment (akshaynadkarni) [inline collections/ansible_collections/osac/service/roles/publish_templates/tests/test.yml]: **(from claude)**, you can only include if you feel this is necessary. 



*Suggestion for stronger assertions* 

The populated test verifies the correct number of PATCH and POST calls, which is good. To strengthen confidence, you could also verify that PATCHes went to the right endpoints (e.g., that the cluster PATCH included the existing ID in the URL path). This would catch a regression where the role PATCHes the wrong endpoint or mixes up resource types.

For example:
```yaml
  - call_log.json | selectattr('method', 'equalto', 'PATCH') | selectattr('path', 'contains', 'cluster_templates') | list | length == 1
  - call_log.json | selectattr('method', 'equalto', 'PATCH') | selectattr('path', 'contains', 'compute_instance_templates') | list | length == 1
  - call_log.json | selectattr('method', 'equalto', 'PATCH') | selectattr('path', 'contains', 'network_classes') | list | length == 1
```

**Review by akshaynadkarni — APPROVED** [review]
Code LGTM. I've left a few comments/suggestions based on my review using Claude's help. None of these are blockers. You can decide whether/not to address them. 

Thanks for adding the tests 🙏 . 

You are good to go once you address the checks.

**Review by akshaynadkarni — APPROVED** [review]
LGTM
